runtime/storage_test.go

```diff
@@ -2057,3 +2058,64 @@ pub contract TopShot: NonFungibleToken {
 
 	require.Equal(t, 0, contractValueReads)
 }
+
+func TestRuntimeStorageSaveCapability(t *testing.T) {
+
+	t.Parallel()
+
+	runtime := NewInterpreterRuntime()
+
+	storage := newTestStorage(nil, nil)
+
+	signer := common.BytesToAddress([]byte{0x42})
+
+	runtimeInterface := &testRuntimeInterface{
+		storage: storage,
+		getSigningAccounts: func() ([]Address, error) {
+			return []Address{signer}, nil
+		},
+	}
+
+	nextTransactionLocation := newTransactionLocationGenerator()
+
+	// Store a capability
+
+	for _, domain := range []common.PathDomain{
+		common.PathDomainPrivate,
+		common.PathDomainPublic,
+	} {
+
+		for typeDescription, typeArgument := range map[string]string{
+			"Untyped": "",
+			"Typed":   "<&Int>",
+		} {
+
+			t.Run(fmt.Sprintf("%s %s", domain.Identifier(), typeDescription), func(t *testing.T) {
+
+				err := runtime.ExecuteTransaction(
+					Script{
+						Source: []byte(fmt.Sprintf(
+							`
+                              transaction {
+                                  prepare(signer: AuthAccount) {
+                                      let cap = signer.getCapability%[1]s(/%[2]s/test)
+                                      signer.save(cap, to: /storage/test%[3]s%[2]s)
+                                  }
+                              }
+			                `,
+							typeArgument,
+							domain.Identifier(),
+							typeDescription,
+						)),
+					},
+					Context{
+						Interface: runtimeInterface,
+						Location:  nextTransactionLocation(),
+					},
+				)
+				require.NoError(t, err)
+
```

Comment:
would it be also good to assert whether the value is in the storage?

Reply:
Good idea! Added that in 0d52b57b02f5aa14eea222f03b37790c81a95e70